tests/common/util.rs

```diff
@@ -1102,19 +1116,50 @@ impl UCommand {
         self
     }
 
+    // TODO: Accept a parameter `delay` which returns delayed from this method. Most use cases are
+    // with some kind of post delay. Without any delay, the output may be empty because we return
+    // immediately. Most of the time a delay of 1ms was already sufficient.
+    // TODO: rename this method after refactoring the tests to run_no_wait and merge with it
+    // TODO: use plain `self` instead of `&mut self`. UCommand is then consumed what's actually more
+    // honest to the user of the run functions and helps to fix the deadlocks with piped stdin
+    pub fn run_no_wait_child(&mut self) -> UChild {
+        let child = self.run_no_wait();
+        UChild::new(
+            child,
+            self.bin_path.clone(),
+            self.util_name.clone(),
+            self.tmpd.clone(),
+            self.output.take(),
+        )
+    }
+
     /// Spawns the command, feeds the stdin if any, and returns the
-    /// child process immediately.
+    /// child process immediately. Do not use this method directly
+    /// if you want to have stderr redirected to stdout. Use
+    /// [`UCommand::run_no_wait_stderr_to_stdout`] instead.
     pub fn run_no_wait(&mut self) -> Child {
```

Comment:
Do we still need this one or can it be replace by `run_no_wait_child` and rename that one to `run_no_wait`?

Reply:
This method will be the method which returns UChild in the end. To avoid misunderstandings: I'll keep the run methods like they are right now until all tests are adjusted to use `run_no_wait_child` and UChild. I want to be able to run the tests during the rewrite. After the tests are rewritten I'll change the `run_no_wait` method to return `UChild` and rename `run_no_wait_child` to `run_no_wait` in all tests. The unnecessary run methods can then be removed.

Reply:
That's what I expected, but I just wanted to check. Sounds good!